### ISSUE TITLE
Multiple rewinds on EPW error

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#7913]
Issue overview
--------------
1. User file with two design days and one RunPeriod, but with Run Weather Simulation = No, fails with this error:
```
   ** Severe  ** Multiple rewinds on EPW while searching for first day 1/1/2017
   **  Fatal  ** Error occurred on EPW while searching for first day, stopped at 2005/12/31 1:0 IO Error=-1
```

2. Same file with Run Weather Simulation = Yes, fails with a more meaningful error:
```
   ** Severe  ** GetNextEnvironment: Actual weather runperiod has been entered but weatherfile DATA PERIOD does not have year included in start/end date.
   **   ~~~   ** ...to match the RunPeriod, the DATA PERIOD should be mm/dd/yyyy for both, or
   **   ~~~   ** ...set "Treat Weather as Actual" to "No".
   ** Severe  ** GetNextEnvironment: Runperiod [mm/dd/yyyy] (Start=01/01/2017,End=12/31/2017) requested not within Data Period(s) from Weather File
   **   ~~~   ** Weather Data Period (Start=01/01/<noyear>,End=12/31/<noyear>)
   **  Fatal  ** GetNextEnvironment: Program terminates due to preceding condition.
```

This reveals that the RunPeriod has Treat Weather as Actual = Yes, changing it to No lets both files run.

3. So, case 2 is ok as-is. But case 1 should run wihthout error because it shouldn't even be looking at the epw, right?
Reproduced the problem with 5ZoneAirCooled (attached below). 

### Details
Some additional details for this issue (if relevant):
 - Platform Win64
 - Version of EnergyPlus 9.3.0
 - Helpdesk ticket number 15377

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file attached.
[5ZoneAirCooled-DDdaysOnly-ActualWeather.idf.txt](https://github.com/NREL/EnergyPlus/files/4451539/5ZoneAirCooled-DDdaysOnly-ActualWeather.idf.txt)

 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by jasondegraw** [comment]
@mjwitte I agree that case 1 should run as-is. There's quite a bit of logic surrounding the environment setup, so I suspect there's a path through that doesn't make much sense and somehow the actual weather flag is set to true when it shouldn't matter. You can assign me if you don't wish to kick this particular unwell equine animal.
